### PR TITLE
Backport: ft(VLTCLT-6): Add getUsersByIds

### PR DESCRIPTION
**Pull request by tmacro**
backport to 7.9

### CODE REVIEW

**Review by jonathan-gramain — APPROVED** [review]

File: lib/IAMClient.js [hunk 1]
```diff
@@ -592,6 +592,38 @@ class VaultClient {
...
+     * @param {Object} options - additional arguments
+     * @param {string} options.reqUid - the request UID
+     * @param {werelogs.RequestLogger} [options.logger] - Logger instance
+     * @param{Function} callback - the callback handling the array of objects
+     * and the error, if there is one
+     * @returns{undefined}
+     */
+    getUsersById(userIds, options, callback) {
+        assert(Array.isArray(userIds), 'userIds must be an array');
+        userIds.every(item => assert(typeof item === 'string'),
```
Comment (watsaqat) [inline lib/IAMClient.js:607]: should you also assume item is not an empty string here?

**Review by watsaqat — APPROVED** [review]

File: lib/IAMClient.js (hunk 1 above)
Comment (tmacro) [reply lib/IAMClient.js:607]: I think letting the backend handle an empty string is fine. It could be considered a subset of an invalid id being passed.

**Comment by tmacro** [comment]
ping

**Review by jonathan-gramain — APPROVED** [review]

**Pull request merged**